cleaner.go

```diff
@@ -3,38 +3,42 @@ package rmq
 import "fmt"
 
 type Cleaner struct {
-	connection *redisConnection
+	connection Connection
 }
 
-func NewCleaner(connection *redisConnection) *Cleaner {
+func NewCleaner(connection Connection) *Cleaner {
 	return &Cleaner{connection: connection}
 }
 
 func (cleaner *Cleaner) Clean() error {
-	connectionNames := cleaner.connection.GetConnections()
+	cleanerConnection, ok := cleaner.connection.(*redisConnection)
+	if !ok {
+		return nil
```

Comment:
Shall we return an error here?
```suggestion
		return errors.New("not implemented")
```

Reply:
I'd want the cleaner to work on a `TestConnection` too - so for this not to be an error we'd need to handle it there. 

At this point it feels more appropriate for `Clean` to be a method on `Connection`

I'm not sure it's worth bothering with this and the unintended consequences if we are able to absorb cleaning into the internals of queue altogether...

Reply:
bump @wellle